tests/com/com_nrf24l01_spi_protocol_tests.cpp

```diff
@@ -0,0 +1,111 @@
+#include "com_nrf24l01_spi_protocol.hpp"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using ::testing::_;
+using ::testing::ElementsAre;
+using ::testing::Invoke;
+using ::testing::Return;
+
+namespace {
+class ComNRF24L01SpiProtocolTests : public ::testing::Test {
+ protected:
+  virtual void SetUp() {
+  }
+  spi::CSPin cs_pin_;
+  spi::SPI spi_{cs_pin_};
+  std::unique_ptr<com::NRF24L01SpiProtocol> unit_under_test_;
+};
+
+}  // namespace
+
+types::DriverStatus ConstructTestVector(std::vector<std::uint8_t> &mosi, std::vector<std::uint8_t> &miso) noexcept {
+  miso.at(0) = 0xaa;
+  return types::DriverStatus::OK;
+}
+
+TEST_F(ComNRF24L01SpiProtocolTests, read_register) {
+  std::uint8_t test_reg_addr = 0xfa;
+  unit_under_test_ = std::make_unique<com::NRF24L01SpiProtocol>(spi_);
+  EXPECT_CALL(spi_, Transfer(ElementsAre(com::instruction_word::R_REGISTER | test_reg_addr), ElementsAre(_, _)));
+  unit_under_test_->ReadRegister(test_reg_addr);
+}
+
+TEST_F(ComNRF24L01SpiProtocolTests, write_register_spi_tx_ok) {
+  std::uint8_t test_reg_addr = 0xfa;
+  std::uint8_t test_reg_content = 0xaf;
+  unit_under_test_ = std::make_unique<com::NRF24L01SpiProtocol>(spi_);
+  EXPECT_CALL(spi_, Write(ElementsAre(com::instruction_word::W_REGISTER | test_reg_addr, test_reg_content)));
+  auto retval = unit_under_test_->WriteRegister(test_reg_addr, test_reg_content);
+  EXPECT_EQ(retval, types::DriverStatus::OK);
```

Comment:
Please use ASSERT_EQ when testing for the testcondition. Expected conditions could be skipped by using the correct gtest flag

Reply:
Done.